### PR TITLE
Add 'nix' subcommand

### PR DESCRIPTION
**Pull request by bketelsen**
If merged, this PR will add a nix subcommand to install and manage a nix installation custom built for VanillaOS

### CODE REVIEW

File: cmd/nix_init.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+func NewNixInitCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Initialize nix repository",
+		RunE:  initNix,
+	}
+
+	return cmd
+}
+func initNix(cmd *cobra.Command, args []string) error {
```
Comment (matbme) [inline cmd/nix_init.go:38]: I think the logic part should go inside `core`, maybe something like `core/nix.go`

Comment (bketelsen) [reply cmd/nix_init.go:38]: yes, when it's not `Draft` anymore I'll do lots of cleanup. Agreed.

**Review by mirkobrombin — APPROVED** [review]

**Review by matbme — APPROVED** [review]

File: cmd/nix_init.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+		RunE: initNix,
+	}
+
+	return cmd
+}
+func initNix(cmd *cobra.Command, args []string) error {
+	// prompt for confirmation
+	log.Default().Printf(`This will create a ".nix" folder in your home directory
+and some systemd units to mount that folder at /nix before running the installation
```
Comment (MonsterObserver) [inline cmd/nix_init.go:36]:
```suggestion
and set up some SystemD units to mount that folder at /nix before running the installation
```

File: cmd/nix_init.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	if proceed != "y" {
+		log.Default().Printf("operation canceled at user request")
+		os.Exit(0)
+	}
+	err := core.NixInit()
+	if err != nil {
+		return err
+	}
+	log.Default().Printf("Installation complete. Reboot to use nix.")
```
Comment (MonsterObserver) [inline cmd/nix_init.go:51]:
```suggestion
	log.Default().Printf("Installation complete. Reboot to start using nix.")
```

File: core/nix.go [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+	}
+	// make sure this isn't being run as root
+	if user == "root" {
+		return errors.New("init must not be run as root user")
+	}
+	unitData := UnitData{User: user}
+
+	// prompt for confirmation
+	log.Default().Printf(`This will create a ".nix" folder in your home directory
+and some systemd units to mount that folder at /nix before running the installation
```
Comment (MonsterObserver) [inline core/nix.go:86]:
```suggestion
and set up some SystemD units to mount that folder at /nix before running the installation
```

**Review by MonsterObserver — CHANGES_REQUESTED** [review]
Minor phrasing improvements

**Comment by bketelsen** [comment]
@MonsterObserver thanks for the updates, PTAL?

**Review by MonsterObserver — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]
Kinda late but here you go

**Pull request merged**